src/runtime/refiner.ts

```diff
@@ -805,25 +857,41 @@ interface OperatorInfo {
-  [Op.LTE]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, evalFn: e => e[0] <= e[1], sqlOp: '<='},
-  [Op.GTE]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, evalFn: e => e[0] >= e[1], sqlOp: '>='},
-  [Op.ADD]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, evalFn: (e: number[]) => e[0] + e[1], sqlOp: '+'},
-  [Op.SUB]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, evalFn: e => e[0] - e[1], sqlOp: '-'},
-  [Op.MUL]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, evalFn: e => e[0] * e[1], sqlOp: '*'},
-  [Op.DIV]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, evalFn: e => e[0] / e[1], sqlOp: '/'},
-  [Op.NOT]: {nArgs: 1, argType: Primitive.BOOLEAN,  evalType: Primitive.BOOLEAN, evalFn: e => !e[0], sqlOp: 'NOT'},
-  [Op.NEG]: {nArgs: 1, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, evalFn: e => -e[0], sqlOp: '-'},
-  [Op.EQ]: {nArgs: 2, argType: 'same', evalType: Primitive.BOOLEAN, evalFn: e => e[0] === e[1], sqlOp: '='},
-  [Op.NEQ]: {nArgs: 2, argType: 'same', evalType: Primitive.BOOLEAN, evalFn: e => e[0] !== e[1], sqlOp: '<>'},
+  [Op.AND]: {nArgs: 2, argType: Primitive.BOOLEAN, evalType: Primitive.BOOLEAN, sqlOp: 'AND'},
+  [Op.OR]: {nArgs: 2, argType: Primitive.BOOLEAN, evalType: Primitive.BOOLEAN, sqlOp: 'OR'},
+  [Op.LT]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, sqlOp: '<'},
+  [Op.GT]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, sqlOp: '>'},
+  [Op.LTE]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, sqlOp: '<='},
+  [Op.GTE]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.BOOLEAN, sqlOp: '>='},
+  [Op.ADD]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, sqlOp: '+'},
+  [Op.SUB]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, sqlOp: '-'},
+  [Op.MUL]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, sqlOp: '*'},
+  [Op.DIV]: {nArgs: 2, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, sqlOp: '/'},
+  [Op.NOT]: {nArgs: 1, argType: Primitive.BOOLEAN,  evalType: Primitive.BOOLEAN, sqlOp: 'NOT'},
+  [Op.NEG]: {nArgs: 1, argType: Primitive.NUMBER,  evalType: Primitive.NUMBER, sqlOp: '-'},
+  [Op.EQ]: {nArgs: 2, argType: 'same', evalType: Primitive.BOOLEAN, sqlOp: '='},
+  [Op.NEQ]: {nArgs: 2, argType: 'same', evalType: Primitive.BOOLEAN, sqlOp: '<>'},
+};
+
+const evalTable: Dictionary<(exprs: ExpressionPrimitives[]) => ExpressionPrimitives> = {
+  [Op.AND]: e => e[0] && e[1],
+  [Op.OR]: e => e[0] || e[1],
+  [Op.LT]: e => e[0] < e[1],
+  [Op.GT]: e => e[0] > e[1],
+  [Op.LTE]: e => e[0] <= e[1],
+  [Op.GTE]: e => e[0] >= e[1],
+  [Op.ADD]: e => e[0] + e[1],
+  [Op.SUB]: e => e[0] - e[1],
+  [Op.MUL]: e => e[0] * e[1],
+  [Op.DIV]: e => e[0] / e[1],
+  [Op.NOT]: e => !e[0],
+  [Op.NEG]: e => -e[0],
+  [Op.EQ]: e => e[0] === e[1],
+  [Op.NEQ]: e => e[0] !== e[1],
```

Comment:
Am slightly surprised that the `[]` are necessary.

Reply:
I'm not exactly sure about the specifics either, but I'm following [this interface](https://github.com/PolymerLabs/arcs/blob/master/src/runtime/hot.ts#L73). Not exactly sure why it complains without the ```[]```.

Reply:
For the record, the `[Op.AND]` is a typescript/javascript ism to get the string form of an identifier for use as a key. This is important here because we're storing the functions in a `Dictionary` rather than a map, so the key type has to be string.

Thanks for showing me ragav.